### DISCUSSION TITLE
Is there inbuilt function to get the persisted data?

### DISCUSSION THREAD

**Discussion by maddyahmad** [TanStack/query#7771] [Q&A]
Is there any function, or way to get that persisted data from the storage? 
Because when i refresh the page, it still calls the api. Even though i could prevent that using staletime, but i was just looking for an inbuilt function, so that I can use persited data in another page. Any solution apart from invalidateQueries would work, because if i refresh the page, invalidateQueries gives me undefined.

const ReactQueryProvider = ({ children }) => {
    const [queryClient] = useState(() => new QueryClient({
        defaultOptions: {
            queries: {
                gcTime: 1000 * 60 * 60 * 24, // 24 hours
            },
        },
    }));
    const persister = createSyncStoragePersister({
        storage: window.localStorage,
    })
    return (
        <PersistQueryClientProvider client={queryClient} persistOptions={{ persister }}>{children}</PersistQueryClientProvider>
    );
};
export default ReactQueryProvider;

**Comment by TkDodo** [comment]
you control the storage by passing `storage: window.localStorage`. You can also control the `key`. Not sure what you want to achieve but that means you can just "read" from localstorage under that key. Otherwise, in the `onSuccess` callback of the `PersistQueryClientProvider`, data will be restored already so you can read it with `queryClient.getQueryData` then
